src/pl/senla/hotel/annotations/di/DI.java

```diff
@@ -0,0 +1,41 @@
+package pl.senla.hotel.annotations.di;
+
+import java.lang.reflect.Field;
+import java.util.List;
+import java.util.Map;
+
+@AppComponent
+public class DI {
+
+    private static DI injector;
+    private DI(){}
+
+    public static DI getSingletonInstance() {
```

Comment:
you don't need to create it as a singleton, you can create it with the **new**  keyword, it's ok or make it static as spring boot doing 
![image](https://github.com/dudarenko-dmitry/senla_training/assets/39222770/feba2fab-5c63-4fc0-8d42-a37d4dd8a03d)

Reply:
ok